apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryParameterCollection.Data.cs

```diff
@@ -26,105 +27,109 @@ public sealed partial class BigQueryParameterCollection : DbParameterCollection
-        {
-            throw new NotImplementedException();
-        }
+        public override int IndexOf(string parameterName) => _parameters.FindIndex(x => x.ParameterName.Equals(parameterName));
 
         /// <inheritdoc />
-        public override void RemoveAt(string parameterName)
-        {
-            throw new NotImplementedException();
-        }
+        public override void RemoveAt(string parameterName) => _parameters.RemoveAt(IndexOf(parameterName));
 
         /// <inheritdoc />
         protected override void SetParameter(int index, DbParameter value)
-        {
-            throw new NotImplementedException();
-        }
+            => _parameters[index] = (BigQueryParameter)value;
 
         /// <inheritdoc />
         protected override void SetParameter(string parameterName, DbParameter value)
         {
-            throw new NotImplementedException();
+            int index = IndexOf(parameterName);
+            if (index == -1)
+            {
+                _parameters.Add((BigQueryParameter)value);
+            }
+            else
+            {
+                _parameters[index] = (BigQueryParameter)value;
+            }
         }
 
 #if NET45
         /// <inheritdoc />
-        public override bool IsFixedSize { get; }
+        public override bool IsFixedSize => false;
         /// <inheritdoc />
-        public override bool IsReadOnly { get; }
+        public override bool IsReadOnly => false;
         /// <inheritdoc />
-        public override bool IsSynchronized { get; }
+        public override bool IsSynchronized => false;
 #endif
         /// <inheritdoc />
-        public override object SyncRoot { get; }
+        public override object SyncRoot => _parameters;
 
         /// <inheritdoc />
         public override void Insert(int index, object value)
-        {
-            throw new NotImplementedException();
-        }
+            => _parameters.Insert(index, (BigQueryParameter)value);
 
         /// <inheritdoc />
         public override void Remove(object value)
-        {
-            throw new NotImplementedException();
-        }
+            => _parameters.Remove((BigQueryParameter)value);
 
         /// <inheritdoc />
-        public override void RemoveAt(int index)
-        {
-            throw new NotImplementedException();
-        }
+        public override void RemoveAt(int index) => _parameters.RemoveAt(index);
 
         /// <inheritdoc />
         public override void CopyTo(Array array, int index)
         {
-            throw new NotImplementedException();
+            if (array == null)
+            {
+                throw new ArgumentNullException(nameof(array));
+            }
+            if (array.Length < _parameters.Count + index)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(array), "There is not enough space in the array to copy values.");
+            }
+            foreach (var item in _parameters)
```

Comment:
Can't we use _parameters.CopyTo?

Reply:
no because the method takes an Array type.